### ISSUE TITLE
TTC calculation

### ISSUE THREAD

**Issue by whatgit** [RobertKrajewski/highD-dataset#9]
Hello,

I am wondering how do you calculate the TTC in the dataset? I was assuming that 
`TTC = dhw/(xVelocity - precedingXVelocity)`; where _TTC_ is time-to-collision, _dhw_ is the distance headway, _xVelocity_ is the ego vehicle's velocity, and _precedingXVelocity_ is the preceding vehicle's velocity. 

However, when I check it with the data, it was not correct. Therefore, I am wondering how do you calculate TTC in the dataset?

Please see the examples below.

# Example from dataset_01
## Case 1: vehicle id 2 is following vehicle id 13, here are the parameters in "track_data" of the vehicle id 2:
dhw = 112.62
ttc = -17.38
thw = 3.51
precedingXVelocity= -38.53
xVelocity = -32.06

If we follow the equation above. TTC should be equal to `112.62/(32.06-38.53) = 112.62/-6.47 = -17.40`
(I think minus sign here indicate driving direction so I will ignore it for now)

## Case 2: vehicle id 8 is following vehicle id 2, here are the parameters in "track_data" of the vehicle id 8:
thw = 2.29
dhw = 72.05
ttc = -104.67
precedingXVelocity = -32.04
xVelocity = -31.36

If we follow the equation above. TTC should be equal to `72.05/(31.36-32.04) = 72.05/-0.68 = -105.95`

**Comment by Victorrsl** [comment]
Yes, I have found this problem too, may I ask if the proposer now knows the cause of this deviation?

**Comment by Victorrsl** [comment]
Hi, considering that TTC is the ratio of distance to speed difference, and the speed difference between two vehicles following each other on the road generally fluctuates around 0, so there are bound to be moments when the speed difference value is 0. How did the authors process the data at this point?I analyzed the TTC value when the speed of the leading vehicle and following vehicle is equal, and found that the ratio of distance and TTC fluctuates between -0.009~0.009, therefore, we can conclude that the author in the batch processing of TTC, in order to avoid more than the appearance of the speed difference value of 0, that is, the denominator is 0, the author in the speed difference after the addition of a disturbance between -0.009~0.009, to ensure the stability of the value.

**Comment by Victorrsl** [comment]
Hi, since TTC is the ratio of distance to speed difference, and this difference between two following vehicles generally fluctuates around zero, moments of zero speed difference are inevitable. How did the authors handle such cases? I analyzed the TTC values when the leading and following vehicles had equal speeds and found that the ratio of distance to TTC fluctuates between -0.009 and 0.009. This suggests that, during batch processing, the authors introduced a disturbance between -0.009 and 0.009 to the speed difference to prevent a zero denominator and ensure stability.

**Comment by Victorrsl** [comment]
> Hello,
> 
> I am wondering how do you calculate the TTC in the dataset? I was assuming that `TTC = dhw/(xVelocity - precedingXVelocity)`; where _TTC_ is time-to-collision, _dhw_ is the distance headway, _xVelocity_ is the ego vehicle's velocity, and _precedingXVelocity_ is the preceding vehicle's velocity.
> 
> However, when I check it with the data, it was not correct. Therefore, I am wondering how do you calculate TTC in the dataset?
> 
> Please see the examples below.
> 
> # Example from dataset_01
> ## Case 1: vehicle id 2 is following vehicle id 13, here are the parameters in "track_data" of the vehicle id 2:
> dhw = 112.62 ttc = -17.38 thw = 3.51 precedingXVelocity= -38.53 xVelocity = -32.06
> 
> If we follow the equation above. TTC should be equal to `112.62/(32.06-38.53) = 112.62/-6.47 = -17.40` (I think minus sign here indicate driving direction so I will ignore it for now)
> 
> ## Case 2: vehicle id 8 is following vehicle id 2, here are the parameters in "track_data" of the vehicle id 8:
> thw = 2.29 dhw = 72.05 ttc = -104.67 precedingXVelocity = -32.04 xVelocity = -31.36
> 
> If we follow the equation above. TTC should be equal to `72.05/(31.36-32.04) = 72.05/-0.68 = -105.95`

Hi, since TTC is the ratio of distance to speed difference, and this difference between two following vehicles generally fluctuates around zero, moments of zero speed difference are inevitable. How did the authors handle such cases? I analyzed the TTC values when the leading and following vehicles had equal speeds and found that the ratio of distance to TTC fluctuates between -0.009 and 0.009. This suggests that, during batch processing, the authors introduced a disturbance between -0.009 and 0.009 to the speed difference to prevent a zero denominator and ensure stability.
